codebuild/cd/osx-arm64-build.sh

```diff
@@ -5,9 +5,16 @@ set -ex
 cd `dirname $0`/../..
 
 git submodule update --init
+export GIT_TAG=$(git describe --tags)
 
-mvn -B compile -P mac-arm64
+# use a fix deploy version for platform specific jar
+mvn versions:set -DnewVersion=deploy
+mvn -B install -DskipTests -P mac-arm64
 
 # Copy artifacts to dist
 mkdir -p ../dist
+# Note: as we ran mvn install, the dist has unexpected files, using S3 bucket to transfer artifact instead
```

Comment:
Haven't really dig into why. But the mac-arm64 build is the only job that I saw unexpected files, and it has the same setting as the x64 build. this [log](https://us-east-1.console.aws.amazon.com/codesuite/codebuild/123124136734/projects/aws-crt-java-cd-deploy-snapshot/build/aws-crt-java-cd-deploy-snapshot%3A961e5727-0db8-4984-9493-b249a60959f7/?region=us-east-1) shows that.